drivers/i3c/i3c_mcux.c

```diff
@@ -1131,6 +1131,9 @@ static int mcux_i3c_transfer(const struct device *dev,
 			goto out_xfer_i3c_stop_unlock;
 		}
 
+		/* write back the total number of bytes transferred */
+		*msgs[i].num_xfer = ret;
```

Comment:
This is not a pointer so there is no need for the `*` beforehand.

Reply:
uh... that is a pointer

Reply:
would you prefer it to not be a pointer?

Reply:
if this stays a pointer... I probably should of checked it if it is NULL :(

Reply:
Ah... never mind... I thought `num_xfer` was just an integer.

Though, I think it is probably better for the ones in the structs to _not_ be pointers. The users of these APIs will not need to keep another variable to store this data as it is already in the `i3c_msg` struct.

Reply:
> Ah... never mind... I thought `num_xfer` was just an integer.
> 
> Though, I think it is probably better for the ones in the structs to _not_ be pointers. The users of these APIs will not need to keep another variable to store this data as it is already in the `i3c_msg` struct.

Yeah, the more I think about it. I'd have to agree... PR updated